pkg/cloud/tags/tags.go

```diff
@@ -127,10 +131,12 @@ func WithEKS(eksclient eksiface.EKSAPI) BuilderOption {
 	return func(b *Builder) {
 		b.applyFunc = func(params *infrav1.BuildParams) error {
 			tags := infrav1.Build(*params)
-
 			eksTags := make(map[string]*string, len(tags))
 			for k, v := range tags {
-				eksTags[k] = aws.String(v)
+				// We want to filter out the tag keys that start with `aws:` as they are reserved for internal AWS use.
+				if !strings.HasPrefix(k, "aws:") {
```

Comment:
Maybe make `aws:` as const ? (Will help readability)

Reply:
Nice suggestion! I've introduced a constant, please take another look.

Reply:
Yep LGTM, thanks for taking care of this.